app/Exceptions/ExceptionHandler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions;
+
+use Illuminate\Foundation\Exceptions\Handler;
+use Illuminate\Support\Facades\Crypt;
+use Inertia\Inertia;
+use Symfony\Component\HttpFoundation\Response;
+use Throwable;
+
+class ExceptionHandler extends Handler
+{
+    protected $dontFlash = [
+        "current_password",
+        "password",
+        "password_confirmation",
+    ];
+
+    public function render($request, Throwable $e)
+    {
+        $response = parent::render($request, $e);
+        $statusCode = $response->status();
+
+        app()->setLocale("en");
+
+        $encryptedCookie = $request->cookie("locale");
+
+        if ($encryptedCookie) {
+            if (in_array($encryptedCookie, ["pl", "en"], true)) {
+                app()->setLocale($encryptedCookie);
+            } else {
+                $decryptedCookie = Crypt::decryptString($encryptedCookie);
+
+                $languageCode = substr($decryptedCookie, strpos($decryptedCookie, "|") + 1);
+
+                if (in_array($languageCode, ["pl", "en"], true)) {
+                    app()->setLocale($languageCode);
+                }
+            }
```

Comment:
why we set app language in handler?

Reply:
It's beacuse of i18n. Error pages couldn't see user language but when it's fetched from cookies, decrypted and set - it works.

Reply:
I would like move that into some middleware.